code/game/objects/items/stacks/stack.dm

```diff
@@ -22,7 +22,11 @@
 /obj/item/stack/New(var/loc, var/amount=null)
 	..()
 	if (amount)
-		src.amount = amount
+		if(amount<=max_amount)
+			src.amount = amount
+		else
+			amount = max_amount
```

Comment:
`src.amount = max_amount` , having an arg and a variable of the src use the same name was a shit idea.